crates/test-programs/wasi-sockets-tests/src/bin/tcp_v4.rs

```diff
@@ -15,6 +16,56 @@ fn wait(sub: poll::Pollable) {
     }
 }
 
+fn write(output: streams::OutputStream, bytes: &[u8]) -> (usize, StreamStatus) {
+    let sub = streams::subscribe_to_write_ready(output);
+    let num_wanted = loop {
+        wait(sub);
+        match streams::check_write(output) {
```

Comment:
this is `blocking-check-write` for the common case. Also, no loop is needed because check_write is guaranteed to give Some() after the pollable is ready.